std/exception.d

```diff
@@ -1191,9 +1194,16 @@ if (__traits(isRef, source) || isDynamicArray!S ||
     }
     else static if (isStaticArray!S)
     {
-        foreach (size_t i; 0 .. S.length)
-            if (mayPointTo(source[i], target)) return true;
-        return false;
+        static if (is(S == void[n], size_t n))
+        {
+            return true;
+        }
```

Comment:
We can do more than this in `mayPointTo`. Minimally `return S.sizeof >= typeof(void*).sizeof` as we know that a data chunk smaller than a pointer cannot contain a pointer. Maximally we could interpret every properly-aligned pointer-sized chunk of the static array as if it were a pointer, which is safe since we never actually dereference the pointer but merely compare it to other pointers.

Reply:
Implemented the latter

Reply:
Branch for CTFE?

Reply:
Added a `__ctfe` branch which pessimistically returns `true` because I didn't find a way to implement the actual logic without using a reinterpreting cast.